### ISSUE TITLE
can't upgrade because of 2 different errors

### ISSUE THREAD

**Issue by ahmed-abdelrazek** [plesk/ubuntu20to22#1]
**Describe the bug**
the first is sqlserver in source list
the second something have to do with plesk mail check conditions

**Feedback archive**
```
root@host2:~# ./ubuntu20to22
Doing preparation checks...
( stage Prepare / action add the service 'plesk-dist-upgrade-resume.service' to resume Plesk dist-upgrade af( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:15 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:16 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:17 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:18 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:19 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:20 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:21 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:22 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:23 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:24 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:25 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:26 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:27 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:28 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:29 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:30 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:31 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:32 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:33 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:34 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:35 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:36 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:37 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:38 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:39 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:40 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:41 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:42 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:43 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:44 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:45 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:46 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:47 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:48 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:49 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:50 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:51 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:52 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:53 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:54 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:55 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:56 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:57 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:58 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 00:59 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:00 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:01 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:02 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:03 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:04 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:05 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:06 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:07 / 24( stage Switch repositories / action set up APT repositories to upgrade from 'focal' to 'jammy' ) 01:08 / 24:31
Failed: set up APT repositories to upgrade from 'focal' to 'jammy'. The reason: Command '['/usr/bin/apt-get', 'update', '-y']' returned non-zero exit status 100.

**************************************************************************************
The dist-upgrade process has failed. Here are the last 100 lines of the /var/log/plesk/ubuntu20to22.log file:
**************************************************************************************
2024-03-19 13:11:35,032 - INFO - Running: ['/usr/sbin/plesk', 'bin', 'passwords', '--encrypt']. Output:
2024-03-19 13:11:35,198 - INFO - Command ['/usr/sbin/plesk', 'bin', 'passwords', '--encrypt'] finished successfully
2024-03-19 13:11:35,198 - DEBUG - Action 'turn on the enhanced security mode in Plesk (encrypt passwords)' completed in 0 s with result: ActionResult(state=<ActionState.SUCCESS: 'success'>, info=None, reboot_requested=None, next_phase=None)
2024-03-19 13:11:35,198 - INFO - Success: turn on the enhanced security mode in Plesk (encrypt passwords)
2024-03-19 13:11:35,198 - DEBUG - _track_flow(phase=None, stage='Prepare')
2024-03-19 13:11:35,199 - DEBUG - flow_tracker(phase=None, stage='Prepare')
2024-03-19 13:11:35,199 - DEBUG - ResumeTracker: phase=None, stage='Prepare'
2024-03-19 13:11:35,199 - DEBUG - Writing ResumeData(phase=<Phase.CONVERT: 'convert'>, argv=['./ubuntu20to22'], upgrader_name='Plesk::Ubuntu20to22Upgrader', stage='Prepare') to '/usr/local/psa/var/ubuntu20to22/resume.json'
2024-03-19 13:11:35,199 - DEBUG - Starting the stage 'Switch repositories'
2024-03-19 13:11:35,199 - INFO - Start stage 'Switch repositories'.
2024-03-19 13:11:35,199 - DEBUG - Invoking action 'set up APT repositories to upgrade from 'focal' to 'jammy''
2024-03-19 13:11:35,199 - INFO - Do: set up APT repositories to upgrade from 'focal' to 'jammy'
2024-03-19 13:11:35,200 - INFO - Running: ['/usr/bin/apt-get', 'update', '-y']. Output:
2024-03-19 13:11:35,294 - INFO - Get:1 http://mirrors.linode.com/ubuntu jammy InRelease
2024-03-19 13:11:35,301 - INFO - Get:2 http://mirrors.linode.com/ubuntu jammy-updates InRelease
2024-03-19 13:11:35,301 - INFO - Get:3 http://autoinstall.plesk.com/pool/PSA_18.0.59_14022 jammy InRelease [10.7 kB]
2024-03-19 13:11:35,301 - INFO - Get:4 http://autoinstall.plesk.com/PHP74_17 jammy InRelease [6,129 B]
2024-03-19 13:11:35,301 - INFO - Get:5 http://autoinstall.plesk.com/PHP83_17 jammy InRelease [6,129 B]
2024-03-19 13:11:35,301 - INFO - Hit:6 http://autoinstall.plesk.com/pool/WPB_18.0.55_74 all InRelease
2024-03-19 13:11:35,304 - INFO - Get:7 http://mirrors.linode.com/ubuntu jammy-backports InRelease
2024-03-19 13:11:35,429 - INFO - Get:8 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
2024-03-19 13:11:35,546 - INFO - Get:9 https://packages.microsoft.com/ubuntu/20.04/mssql-server-2022 jammy InRelease [3,592 B]
2024-03-19 13:11:35,583 - INFO - Get:10 http://autoinstall.plesk.com/pool/PSA_18.0.59_14022 jammy/extras amd64 Packages [11.4 kB]
2024-03-19 13:11:35,637 - INFO - Get:11 http://autoinstall.plesk.com/PHP74_17 jammy/all amd64 Packages [9,506 B]
2024-03-19 13:11:35,641 - INFO - Get:12 http://mirrors.linode.com/ubuntu jammy/main i386 Packages [1,040 kB]
2024-03-19 13:11:35,646 - INFO - Get:13 https://esm.ubuntu.com/apps/ubuntu jammy-apps-security InRelease [7,553 B]
2024-03-19 13:11:35,655 - INFO - Get:14 http://mirrors.linode.com/ubuntu jammy/main amd64 Packages [1,395 kB]
2024-03-19 13:11:35,668 - INFO - Get:15 http://mirrors.linode.com/ubuntu jammy/main Translation-en [510 kB]
2024-03-19 13:11:35,673 - INFO - Get:16 http://mirrors.linode.com/ubuntu jammy/main amd64 c-n-f Metadata [30.3 kB]
2024-03-19 13:11:35,714 - INFO - Get:17 http://autoinstall.plesk.com/PHP83_17 jammy/all amd64 Packages [9,126 B]
2024-03-19 13:11:35,724 - INFO - Ign:18 https://packages.microsoft.com/ubuntu/20.04/prod jammy InRelease
2024-03-19 13:11:35,731 - INFO - Get:19 https://esm.ubuntu.com/apps/ubuntu jammy-apps-updates InRelease [7,456 B]
2024-03-19 13:11:35,816 - INFO - Get:20 https://esm.ubuntu.com/infra/ubuntu jammy-infra-security InRelease [7,450 B]
2024-03-19 13:11:35,878 - INFO - Get:21 http://mirrors.linode.com/ubuntu jammy/restricted i386 Packages [30.4 kB]
2024-03-19 13:11:35,879 - INFO - Get:22 http://mirrors.linode.com/ubuntu jammy/restricted amd64 Packages [129 kB]
2024-03-19 13:11:35,881 - INFO - Get:23 http://mirrors.linode.com/ubuntu jammy/restricted Translation-en [18.6 kB]
2024-03-19 13:11:35,881 - INFO - Get:24 http://mirrors.linode.com/ubuntu jammy/restricted amd64 c-n-f Metadata [488 B]
2024-03-19 13:11:35,900 - INFO - Get:25 https://esm.ubuntu.com/infra/ubuntu jammy-infra-updates InRelease [7,449 B]
2024-03-19 13:11:36,099 - INFO - Get:26 http://mirrors.linode.com/ubuntu jammy/universe i386 Packages [7,474 kB]
2024-03-19 13:11:36,112 - INFO - Err:27 https://packages.microsoft.com/ubuntu/20.04/prod jammy Release
2024-03-19 13:11:36,112 - INFO - 404  Not Found [IP: 13.107.213.64 443]
2024-03-19 13:11:36,148 - INFO - Get:28 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [1,272 kB]
2024-03-19 13:11:36,172 - INFO - Get:29 http://mirrors.linode.com/ubuntu jammy/universe amd64 Packages [14.1 MB]
2024-03-19 13:11:36,293 - INFO - Get:30 https://esm.ubuntu.com/apps/ubuntu jammy-apps-security/main i386 Packages [43.2 kB]
2024-03-19 13:11:36,311 - INFO - Get:31 http://mirrors.linode.com/ubuntu jammy/universe Translation-en [5,652 kB]
2024-03-19 13:11:36,365 - INFO - Get:32 http://mirrors.linode.com/ubuntu jammy/universe amd64 c-n-f Metadata [286 kB]
2024-03-19 13:11:36,369 - INFO - Get:33 http://mirrors.linode.com/ubuntu jammy/multiverse amd64 Packages [217 kB]
2024-03-19 13:11:36,441 - INFO - Get:34 https://esm.ubuntu.com/apps/ubuntu jammy-apps-security/main amd64 Packages [65.7 kB]
2024-03-19 13:11:36,525 - INFO - Get:35 https://esm.ubuntu.com/apps/ubuntu jammy-apps-updates/main amd64 Packages [479 B]
2024-03-19 13:11:36,545 - INFO - Get:36 http://security.ubuntu.com/ubuntu jammy-security/main i386 Packages [426 kB]
2024-03-19 13:11:36,560 - INFO - Get:37 http://mirrors.linode.com/ubuntu jammy/multiverse i386 Packages [112 kB]
2024-03-19 13:11:36,561 - INFO - Get:38 http://mirrors.linode.com/ubuntu jammy/multiverse Translation-en [112 kB]
2024-03-19 13:11:36,561 - INFO - Get:39 http://security.ubuntu.com/ubuntu jammy-security/main Translation-en [228 kB]
2024-03-19 13:11:36,563 - INFO - Get:40 http://mirrors.linode.com/ubuntu jammy/multiverse amd64 c-n-f Metadata [8,372 B]
2024-03-19 13:11:36,564 - INFO - Get:41 http://mirrors.linode.com/ubuntu jammy-updates/main i386 Packages [592 kB]
2024-03-19 13:11:36,570 - INFO - Get:42 http://security.ubuntu.com/ubuntu jammy-security/main amd64 c-n-f Metadata [11.4 kB]
2024-03-19 13:11:36,571 - INFO - Get:43 http://mirrors.linode.com/ubuntu jammy-updates/main amd64 Packages [1,490 kB]
2024-03-19 13:11:36,571 - INFO - Get:44 http://security.ubuntu.com/ubuntu jammy-security/restricted i386 Packages [33.7 kB]
2024-03-19 13:11:36,572 - INFO - Get:45 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [1,572 kB]
2024-03-19 13:11:36,586 - INFO - Get:46 http://mirrors.linode.com/ubuntu jammy-updates/main Translation-en [288 kB]
2024-03-19 13:11:36,589 - INFO - Get:47 http://mirrors.linode.com/ubuntu jammy-updates/main amd64 c-n-f Metadata [16.1 kB]
2024-03-19 13:11:36,657 - INFO - Get:48 http://security.ubuntu.com/ubuntu jammy-security/restricted Translation-en [263 kB]
2024-03-19 13:11:36,665 - INFO - Get:49 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 c-n-f Metadata [520 B]
2024-03-19 13:11:36,666 - INFO - Get:50 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [851 kB]
2024-03-19 13:11:36,691 - INFO - Get:51 http://security.ubuntu.com/ubuntu jammy-security/universe i386 Packages [599 kB]
2024-03-19 13:11:36,712 - INFO - Get:52 https://esm.ubuntu.com/infra/ubuntu jammy-infra-security/main amd64 Packages [479 B]
2024-03-19 13:11:36,712 - INFO - Get:53 https://esm.ubuntu.com/infra/ubuntu jammy-infra-updates/main amd64 Packages [479 B]
2024-03-19 13:11:36,774 - INFO - Get:54 http://security.ubuntu.com/ubuntu jammy-security/universe Translation-en [162 kB]
2024-03-19 13:11:36,776 - INFO - Get:55 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 c-n-f Metadata [16.8 kB]
2024-03-19 13:11:36,777 - INFO - Get:56 http://security.ubuntu.com/ubuntu jammy-security/multiverse i386 Packages [1,356 B]
2024-03-19 13:11:36,777 - INFO - Get:57 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [37.1 kB]
2024-03-19 13:11:36,778 - INFO - Get:58 http://security.ubuntu.com/ubuntu jammy-security/multiverse Translation-en [7,476 B]
2024-03-19 13:11:36,778 - INFO - Get:59 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 c-n-f Metadata [260 B]
2024-03-19 13:11:36,790 - INFO - Get:60 http://mirrors.linode.com/ubuntu jammy-updates/restricted i386 Packages [35.3 kB]
2024-03-19 13:11:36,791 - INFO - Get:61 http://mirrors.linode.com/ubuntu jammy-updates/restricted amd64 Packages [1,605 kB]
2024-03-19 13:11:36,807 - INFO - Get:62 http://mirrors.linode.com/ubuntu jammy-updates/restricted Translation-en [268 kB]
2024-03-19 13:11:36,811 - INFO - Get:63 http://mirrors.linode.com/ubuntu jammy-updates/restricted amd64 c-n-f Metadata [520 B]
2024-03-19 13:11:36,812 - INFO - Get:64 http://mirrors.linode.com/ubuntu jammy-updates/universe i386 Packages [696 kB]
2024-03-19 13:11:36,818 - INFO - Get:65 http://mirrors.linode.com/ubuntu jammy-updates/universe amd64 Packages [1,058 kB]
2024-03-19 13:11:36,826 - INFO - Get:66 http://mirrors.linode.com/ubuntu jammy-updates/universe Translation-en [239 kB]
2024-03-19 13:11:36,829 - INFO - Get:67 http://mirrors.linode.com/ubuntu jammy-updates/universe amd64 c-n-f Metadata [22.1 kB]
2024-03-19 13:11:36,830 - INFO - Get:68 http://mirrors.linode.com/ubuntu jammy-updates/multiverse amd64 Packages [42.1 kB]
2024-03-19 13:11:37,022 - INFO - Get:69 http://mirrors.linode.com/ubuntu jammy-updates/multiverse i386 Packages [4,160 B]
2024-03-19 13:11:37,023 - INFO - Get:70 http://mirrors.linode.com/ubuntu jammy-updates/multiverse Translation-en [10.1 kB]
2024-03-19 13:11:37,024 - INFO - Get:71 http://mirrors.linode.com/ubuntu jammy-updates/multiverse amd64 c-n-f Metadata [472 B]
2024-03-19 13:11:37,035 - INFO - Get:72 http://mirrors.linode.com/ubuntu jammy-backports/main i386 Packages [59.2 kB]
2024-03-19 13:11:37,036 - INFO - Get:73 http://mirrors.linode.com/ubuntu jammy-backports/main amd64 Packages [67.1 kB]
2024-03-19 13:11:37,037 - INFO - Get:74 http://mirrors.linode.com/ubuntu jammy-backports/main Translation-en [11.0 kB]
2024-03-19 13:11:37,038 - INFO - Get:75 http://mirrors.linode.com/ubuntu jammy-backports/main amd64 c-n-f Metadata [388 B]
2024-03-19 13:11:37,038 - INFO - Get:76 http://mirrors.linode.com/ubuntu jammy-backports/restricted amd64 c-n-f Metadata [116 B]
2024-03-19 13:11:37,039 - INFO - Get:77 http://mirrors.linode.com/ubuntu jammy-backports/universe i386 Packages [17.2 kB]
2024-03-19 13:11:37,039 - INFO - Get:78 http://mirrors.linode.com/ubuntu jammy-backports/universe amd64 Packages [28.4 kB]
2024-03-19 13:11:37,040 - INFO - Get:79 http://mirrors.linode.com/ubuntu jammy-backports/universe Translation-en [16.2 kB]
2024-03-19 13:11:37,040 - INFO - Get:80 http://mirrors.linode.com/ubuntu jammy-backports/universe amd64 c-n-f Metadata [644 B]
2024-03-19 13:11:37,041 - INFO - Get:81 http://mirrors.linode.com/ubuntu jammy-backports/multiverse amd64 c-n-f Metadata [116 B]
2024-03-19 13:12:29,294 - INFO - Reading package lists...
2024-03-19 13:12:29,316 - INFO - E: The repository 'https://packages.microsoft.com/ubuntu/20.04/prod jammy Release' does not have a Release file.
2024-03-19 13:12:29,318 - ERROR - Command ['/usr/bin/apt-get', 'update', '-y'] failed with return code 100
2024-03-19 13:12:29,318 - ERROR - Failed: set up APT repositories to upgrade from 'focal' to 'jammy'. The reason: Command '['/usr/bin/apt-get', 'update', '-y']' returned non-zero exit status 100.
2024-03-19 13:12:29,738 - DEBUG - Going to write json '/usr/local/psa/var/ubuntu20to22/actions.json' with new data

**************************************************************************************
The dist-ugrade process has failed. See the /var/log/plesk/ubuntu20to22.log file for more information.
The last 100 lines of the file are shown above.
For assistance, call 'ubuntu20to22 --prepare-feedback' and follow the instructions.
**************************************************************************************
root@host2:~# ./ubuntu20to22 --prepare-feedback

**************************************************************************************
The feedback archive is ready. You can find it here: ubuntu20to22_feedback.zip
For further assistance, create an issue in our GitHub repository - https://github.com/plesk/ubuntu20to22/issues.
Please attach the feedback archive to the created issue and provide as much information about the problem as you can.
**************************************************************************************
root@host2:~# sudo cp /etc/apt/sources.list /etc/apt/sources.list.bak
root@host2:~# sudo nano /etc/apt/sources.list
root@host2:~# ./ubuntu20to22
Doing preparation checks...
Preparation checks failed: Exception during checking of required pre-conversion condition 'check Plesk components state: -mailman'
Traceback (most recent call last):
  File "/root/ubuntu20to22/pleskdistup/common/src/action.py", line 463, in make_checks
    if not check.do_check():
  File "/root/ubuntu20to22/pleskdistup/common/src/action.py", line 435, in do_check
    return self._do_check()
  File "/root/ubuntu20to22/pleskdistup/actions/plesk.py", line 394, in _do_check
    comp_list = plesk.list_installed_components()
  File "/root/ubuntu20to22/pleskdistup/common/src/plesk.py", line 137, in list_installed_components
    proc = subprocess.run(
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/sbin/plesk', 'installer', '--select-release-current', '--show-components']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/root/ubuntu20to22/pleskdistup/main.py", line 107, in required_conditions_satisfied
    failed_checks = check_flow.make_checks()
  File "/root/ubuntu20to22/pleskdistup/common/src/action.py", line 468, in make_checks
    raise RuntimeError(f"Exception during checking of required pre-conversion condition {check.name!r}") from e
RuntimeError: Exception during checking of required pre-conversion condition 'check Plesk components state: -mailman'

Conversion can't be performed due to the problems noted above

```

**Comment by kpushkaryov** [comment]
Could you please attach the feedback archive `ubuntu20to22_feedback.zip` you prepared by `./ubuntu20to22 --prepare-feedback`?
Also please run `/usr/sbin/plesk installer --select-release-current --show-components` on the server and show the result.

**Comment by ahmed-abdelrazek** [comment]
> Could you please attach the feedback archive `ubuntu20to22_feedback.zip` you prepared by `./ubuntu20to22 --prepare-feedback`? 

no, sorry

> Also please run `/usr/sbin/plesk installer --select-release-current --show-components` on the server and show the result.

```
Detecting installed product components.
  panel             [up2date] - Plesk
  bind              [up2date] - BIND DNS server
  postgresql        [up2date] - PostgreSQL server
  fail2ban          [up2date] - Fail2Ban
  l10n              [up2date] - All language localization for Plesk
  git               [up2date] - Git
  resctrl           [install] - Resource Controller (Cgroups)
  pmm               [install] - Plesk Migrator
  sitebuilder       [install] - Web Presence Builder
  mysqlgroup        [up2date] - MySQL server
  horde             [install] - Horde
  roundcube         [install] - Roundcube
  kav               [install] - Kaspersky Anti-Virus
  drweb             [install] - Plesk Premium Antivirus
  spamassassin      [install] - SpamAssassin
  postfix           [up2date] - Postfix
  qmail             [install] - Qmail
  msmtp             [install] - MSMTP (relay only)
  dovecot           [up2date] - Dovecot
  courier           [install] - Courier
  proftpd           [up2date] - ProFTPD
  webalizer         [up2date] - Webalizer
  awstats           [up2date] - AWStats
  modsecurity       [up2date] - ModSecurity
  passenger         [up2date] - Phusion Passenger server
  ruby              [install] - Ruby support
  nodejs            [up2date] - NodeJS support
  gems-pre          [install] - Tools required for building Ruby gems
  mod_fcgid         [up2date] - mod_fcgid
  mod_perl          [install] - mod_perl
  mod_python        [install] - mod_python
  webservers        [up2date] - Apache
  php8.3            [up2date] - PHP 8.3
  php8.2            [install] - PHP 8.2
  php8.1            [install] - PHP 8.1
  php8.0            [install] - PHP 8.0 (outdated)
  php7.4            [up2date] - PHP 7.4 (outdated)
  phpgroup          [install] - PHP from OS vendor
  nginx             [up2date] - Nginx web server
  config-troubleshooter[install] - Plesk Web Server Configuration Troubleshooter
  psa-firewall      [up2date] - Plesk Firewall
  wp-toolkit        [up2date] - WP Toolkit
  advisor           [up2date] - Advisor
  xovi              [up2date] - SEO Toolkit
  imunifyav         [up2date] - ImunifyAV
  sslit             [up2date] - SSL It!
  letsencrypt       [up2date] - Let's Encrypt
  repair-kit        [up2date] - Repair Kit
  composer          [up2date] - PHP Composer
  monitoring        [up2date] - Monitoring
  log-browser       [up2date] - Log Browser
  ssh-terminal      [up2date] - SSH Terminal
  site-import       [up2date] - Site Import
  sitejet           [up2date] - Sitejet Builder
```

and i guess + .net toolkit extention (.net 6 7 8) and sqlserver

**Comment by kpushkaryov** [comment]
What does `plesk version` return?

**Comment by ahmed-abdelrazek** [comment]
> What does plesk version return?

![image](https://github.com/plesk/ubuntu20to22/assets/7858946/c52bc2fb-3c4d-456d-b1a3-5129278aa4b6)
i think my ubuntu updated in the background anyway

> Also, what does `find /etc/apt/ \( -name '*.list' -o -name '*.sources' \) -ls -exec cat {} \; -exec echo -e '\n\n' \;` show?

```
root@host2:~# find /etc/apt/ \( -name '*.list' -o -name '*.sources' \) -ls -exec cat {} \; -exec echo -e '\n\n' \;
   131297      4 -rw-r--r--   1 root     root          139 Mar 20 17:13 /etc/apt/sources.list.d/plesk-ext-dot-net.list
deb [arch=amd64,armhf,arm64 signed-by=/etc/apt/keyrings/plesk-ext-dot-net.gpg] https://packages.microsoft.com/ubuntu/22.04/prod jammy main



   140791      4 -rw-r--r--   1 root     root           89 Mar 20 17:13 /etc/apt/sources.list.d/microsoft-prod.list
deb [arch=amd64,arm64,armhf] https://packages.microsoft.com/ubuntu/20.04/prod focal main



   131250      4 -rw-r--r--   1 root     root          266 Mar 20 17:13 /etc/apt/sources.list.d/ubuntu-esm-apps.list
deb https://esm.ubuntu.com/apps/ubuntu jammy-apps-security main
# deb-src https://esm.ubuntu.com/apps/ubuntu jammy-apps-security main
deb https://esm.ubuntu.com/apps/ubuntu jammy-apps-updates main
# deb-src https://esm.ubuntu.com/apps/ubuntu jammy-apps-updates main



   131277      4 -rw-r--r--   1 root     root          274 Mar 20 17:13 /etc/apt/sources.list.d/ubuntu-esm-infra.list
deb https://esm.ubuntu.com/infra/ubuntu jammy-infra-security main
# deb-src https://esm.ubuntu.com/infra/ubuntu jammy-infra-security main
deb https://esm.ubuntu.com/infra/ubuntu jammy-infra-updates main
# deb-src https://esm.ubuntu.com/infra/ubuntu jammy-infra-updates main



   131279      4 -rw-r--r--   1 root     root         2955 Mar 20 17:13 /etc/apt/sources.list
# See http://help.ubuntu.com/community/UpgradeNotes for how to upgrade to
# newer versions of the distribution.
deb http://mirrors.linode.com/ubuntu/ jammy main restricted
# deb-src http://mirrors.linode.com/ubuntu/ jammy main restricted

## Major bug fix updates produced after the final release of the
## distribution.
deb http://mirrors.linode.com/ubuntu/ jammy-updates main restricted
# deb-src http://mirrors.linode.com/ubuntu/ jammy-updates main restricted

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team. Also, please note that software in universe WILL NOT receive any
## review or updates from the Ubuntu security team.
deb http://mirrors.linode.com/ubuntu/ jammy universe
# deb-src http://mirrors.linode.com/ubuntu/ jammy universe
deb http://mirrors.linode.com/ubuntu/ jammy-updates universe
# deb-src http://mirrors.linode.com/ubuntu/ jammy-updates universe

## N.B. software from this repository is ENTIRELY UNSUPPORTED by the Ubuntu
## team, and may not be under a free licence. Please satisfy yourself as to
## your rights to use the software. Also, please note that software in
## multiverse WILL NOT receive any review or updates from the Ubuntu
## security team.
deb http://mirrors.linode.com/ubuntu/ jammy multiverse
# deb-src http://mirrors.linode.com/ubuntu/ jammy multiverse
deb http://mirrors.linode.com/ubuntu/ jammy-updates multiverse
# deb-src http://mirrors.linode.com/ubuntu/ jammy-updates multiverse

## N.B. software from this repository may not have been tested as
## extensively as that contained in the main release, although it includes
## newer versions of some applications which may provide useful features.
## Also, please note that software in backports WILL NOT receive any review
## or updates from the Ubuntu security team.
deb http://mirrors.linode.com/ubuntu/ jammy-backports main restricted universe multiverse
# deb-src http://mirrors.linode.com/ubuntu/ jammy-backports main restricted universe multiverse

## Uncomment the following two lines to add software from Canonical's
## 'partner' repository.
## This software is not part of Ubuntu, but is offered by Canonical and the
## respective vendors as a service to Ubuntu users.
# deb http://archive.canonical.com/ubuntu jammy partner
# deb-src http://archive.canonical.com/ubuntu jammy partner

deb http://security.ubuntu.com/ubuntu/ jammy-security main restricted
# deb-src http://security.ubuntu.com/ubuntu/ jammy-security main restricted
deb http://security.ubuntu.com/ubuntu/ jammy-security universe
# deb-src http://security.ubuntu.com/ubuntu/ jammy-security universe
deb http://security.ubuntu.com/ubuntu/ jammy-security multiverse
# deb-src http://security.ubuntu.com/ubuntu/ jammy-security multiverse
# deb [arch=arm64,amd64,armhf] https://packages.microsoft.com/ubuntu/20.04/mssql-server-2022 jammy main
# deb-src [arch=arm64,amd64,armhf] https://packages.microsoft.com/ubuntu/20.04/mssql-server-2022 jammy main



root@host2:~#
```

**Comment by kpushkaryov** [comment]
Both errors were caused by the incorrect APT source `https://packages.microsoft.com/ubuntu/20.04/prod jammy`. It was created when ubuntu20to22 replaced old distribution codename "focal" with new "jammy" without changing the version.

As a workaround, you may want to switch such sources manually to the new distribution before running ubuntu20to22.

**Comment by SandakovMM** [comment]
I think the issue should be resolved by #5 because 20 will automatically change to 22 now.
The change available since [v1.0.1](https://github.com/plesk/ubuntu20to22/releases/tag/v1.0.1)
